### PR TITLE
Skip clusterNode validation for runSh jobs

### PR DESCRIPTION
**Pull request by shrivara**
Plus, remove some more isServiceNode references

https://github.com/Shippable/reqProc/issues/111
https://github.com/Shippable/reqProc/issues/110

### CODE REVIEW

File: workflows/runSh.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ var cleanup = require('../job/cleanup.js');
 var updateStatus = require('../job/updateStatus.js');
 
 function runSh(externalBag, callback) {
+  // At this point we have started processing the runSh and we do not want
+  // the cluster node validation to affect the build.
+  global.config.isProcessingRunSh = true;
```
Comment (shrivara) [inline workflows/runSh.js]: This deviates from the original design, but is a much simpler way of figuring out if the node is processing a job currently.

Comment (ric03uec) [reply workflows/runSh.js]: can we just call it `isJobInProgress` or `isProcessingJob` to be more clear on what this is

Comment (shrivara) [reply workflows/runSh.js]: I didn't want it to be confused with a CI job.

Comment (shrivara) [reply workflows/runSh.js]: `isProcessingRunShJob` maybe?

**Pull request merged**